ext/js/language/text-scanner.js

```diff
@@ -259,8 +262,10 @@ class TextScanner extends EventDispatcher {
 
     setCurrentTextSource(textSource) {
         this._textSourceCurrent = textSource;
-        if (this._selectText) {
+        if (this._selectText && this._userHasNotSelectedAnythingManually) {
+            this._yomichanIsChangingTextSelectionNow = true;
             this._textSourceCurrent.select();
+            setTimeout(() => this._yomichanIsChangingTextSelectionNow = false, 0);
```

Comment:
Probably want to have this callback defined in the constructor so it doesn't have to create a new closure every time it's invoked.

Reply:
I can do that, but I was assuming that JS VM should be sufficiently optimizing this sort of thing. It seems that closures are extremely common in JS. Do you think that trying to manually optimize this is worthwhile?

Reply:
This is something I've done for a few high-traffic code paths, but doing some testing with a similar case doesn't show much of a difference either way. So at worst it would not affect anything, but overall it probably doesn't matter.

Reply:
When I learned of the existence of profiling flame charts, I stopped trying to optimize minor things like that. The bottlenecks were never where I thought they would be!